app/page.tsx

```diff
@@ -75,16 +82,18 @@ export default function Home() {
   function renderButton(data: any) {
     const currentTime = new Date();
     const sessionTimeStr = formatSessionTime(data.start_time);
+    const sessionEndTimeStr = formatSessionTime(data.end_time);
     const currentTimeStr = formatCurrentTime(currentTime.toISOString());
 
     const sessionTime = new Date(`2000-01-01T${sessionTimeStr}`);
+    const sessionEndTime = new Date(`2000-01-01T${sessionEndTimeStr}`);
     const currentTimeObj = new Date(`2000-01-01T${currentTimeStr}`);
     const timeDifference = (sessionTime.getTime() - currentTimeObj.getTime()) / (1000 * 60);
 
-    if (data.platform === 'meet') {
-      if (timeDifference <= 5) {
+    if (data.session && data.session.platform === 'meet') {
+      if (timeDifference <= 5 && sessionEndTime.getTime() > currentTimeObj.getTime()) {
```

Comment:
maybe call these variables something else for clear understanding. 

const minutesUntilSessionStart = (sessionStartTime.getTime() - currentTimeObj.getTime()) / (1000 * 60);

const hasSessionNotEnded = sessionEndTime.getTime() > currentTimeObj.getTime()

Reply:
Made suggested changes!